src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BytesKmerizer.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Iterator over successive Kmers from a byte[].
+ * Bytes are interpreted as ASCII characters.
+ * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
+ *
+ * Created by tsharpe on 12/18/15.
+ */
+public class BytesKmerizer implements Iterator<Kmer>
+{
+    private final byte[] seq;
+    private final int kSize;
+    private int idx;
+    private Kmer kmer;
+
+    public BytesKmerizer( final byte[] someSeq, final int someKSize )
+    {
+        seq = someSeq;
+        kSize = someKSize;
+        idx = 0;
+        kmer = new Kmer(kSize);
+        fill(0);
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+        return kmer != null;
+    }
+
+    @Override
+    public Kmer next()
+    {
+        if ( kmer == null ) throw new NoSuchElementException("Kmerization sequence exhausted.");
+        final Kmer result = kmer;
+        fill(kSize -1);
+        return result;
+    }
+
+    public static Kmer toKmer( final byte[] seq )
+    {
+        final BytesKmerizer bk = new BytesKmerizer(seq,seq.length);
+        if ( !bk.hasNext() )
+            throw new IllegalArgumentException("Byte array contained illegal characters");
+        return bk.next();
+    }
+
+    private void fill( int validBaseCount )
+    {
+        Kmer tmpKmer = kmer;
+        kmer = null;
+
+        final int len = seq.length;
+        while ( idx < len )
```

Comment:
No, you only do that for the first Kmer.  Subsequent calls will go through the loop just once, replacing the previous kmer with the next one (assuming that the next character is a valid base).  Should we worry about the K-1 temporary kmers in producing the first one?  Nah.  K is a small constant.  The JVM is optimized for precisely this case.  And it keeps the code compact and non-redundant.